media/js/firefox/australis/australis-uitour.js

```diff
@@ -56,14 +56,71 @@ if (typeof Mozilla == 'undefined') {
         return id;
     }
 
+    var notificationListener = null;
+    function _notificationListener(event) {
+        if (typeof event.detail != 'object')
+            return;
+        if (typeof notificationListener != 'function')
+            return;
+
+        notificationListener(event.detail.event, event.detail.params);
+    }
+
     Mozilla.UITour.DEFAULT_THEME_CYCLE_DELAY = 10 * 1000;
 
+    Mozilla.UITour.CONFIGNAME_SYNC = 'sync';
+    Mozilla.UITour.CONFIGNAME_AVAILABLETARGETS = 'availableTargets';
+
+    Mozilla.UITour.ping = function(callback) {
+        var data = {};
+        if (callback) {
+            data.callbackID = _waitForCallback(callback);
+        }
+        _sendEvent('ping', data);
+    };
+
+    Mozilla.UITour.observe = function(listener, callback) {
+        notificationListener = listener;
+
+        if (listener) {
+            document.addEventListener('mozUITourNotification',
+                                                                _notificationListener);
+            Mozilla.UITour.ping(callback);
+        } else {
+            document.removeEventListener('mozUITourNotification',
+                                                                     _notificationListener);
+        }
+    };
+
     Mozilla.UITour.registerPageID = function(pageID) {
         _sendEvent('registerPageID', {
             pageID: pageID
         });
     };
 
+    Mozilla.UITour.showHeartbeat = function(message, thankyouMessage, flowId, engagementURL,
```

Comment:
Instead of pasting the whole contents of this file from mozilla central, I would only add what we need. So far I've only been including stuff that has actively been used on bedrock (and we have documentation for). Things such as `showHeartBeat()` we're never intended to be used in the context of mozorg, but from the browser itself (it is called from an invisible tab I believe).